bodhi/server/push.py

```diff
@@ -94,44 +116,37 @@ def push(username, password, cert_prefix, **kwargs):
 
-                updates.append(update.title)
-                for build in update.builds:
-                    releases[update.release.name][request].append(build.nvr)
-            while resp.page < resp.pages:
-                resp = client.query(request=request, page=resp.page + 1, **kwargs)
-                for update in resp.updates:
-                    updates.append(update.title)
-                    for build in update.builds:
-                        releases[update.release.name][request].append(build.nvr)
-
-            # Write out a file that releng uses to pass to sigul for signing
-            # TODO: in the future we should integrate signing into the workflow
-            for release in releases:
-                output_filename = request.title() + '-' + release
-                click.echo(output_filename + '\n==========')
-                with file(output_filename, 'w') as out:
-                    for build in releases[release][request]:
-                        out.write(build + '\n')
-                        click.echo(build)
-                click.echo('')
-
-    doit = raw_input('Push these %d updates? (y/n)' % len(updates)).lower().strip()
-    if doit == 'y':
-        click.echo('\nSending masher.start fedmsg')
-        # Because we're a script, we want to send to the fedmsg-relay,
-        # that's why we say active=True
-        bodhi.server.notifications.init(active=True, cert_prefix=cert_prefix)
-        bodhi.server.notifications.publish(
-            topic='masher.start',
-            msg=dict(
-                updates=list(updates),
-                resume=resume,
-                agent=username,
-            ),
-            force=True,
-        )
-    else:
-        click.echo('\nAborting push')
+                # Skip unsigned updates (this checks that all builds in the update are signed)
+                if not update.signed:
+                    click.echo('Warning: %s has unsigned builds and has been skipped' %
+                               update.title)
+                    continue
+
+                updates.append(update)
+
+        doit = raw_input('Push these %d updates? (y/n)' % len(updates)).lower().strip()
+        if doit == 'y':
+            click.echo('\nLocking updates...')
+            for update in updates:
+                update.locked = True
+                update.date_locked = datetime.utcnow()
+
+            click.echo('\nSending masher.start fedmsg')
+            # Because we're a script, we want to send to the fedmsg-relay,
+            # that's why we say active=True
+            bodhi.server.notifications.init(active=True, cert_prefix=cert_prefix)
+            bodhi.server.notifications.publish(
+                topic='masher.start',
+                msg=dict(
+                    updates=list([update.title for update in updates]),
+                    resume=resume,
+                    agent=username,
+                ),
+                force=True,
+            )
```

Comment:
Could we move this fedmsg to be outside of the transaction so that the transaction is committed before the message is sent? It could be bad if the transaction.commit() fails and we sent the message already, or if the message is received and processed before the transaction is committed somehow.